### ISSUE TITLE
Uncaught Error: The specified module could not be found./\?\~\AppData\Local\atom\app-1.40.0\res...

### ISSUE THREAD

**Issue by kuovonne** [atom/atom#19784]
Please forgive my mistakes. I am completely new to Atom, GitHub, and coding in general.
I just started using Atom and it was working okay when I last used it a few days ago, but today I get a big red error whenever I press a key on the keyboard. I had no idea what to do so I clicked the "Create issue on atom/atom" button in the error message and this screen appeared.

[Enter steps to reproduce:]

1. Click the Atom button on the Windows taskbar to start Atom.
2. Press a key on the keyboard.

**Atom**: 1.40.0 ia32
**Electron**: 3.1.10
**OS**: Microsoft Windows 10 Home
**Thrown From**: Atom Core

### Stack Trace

Uncaught Error: The specified module could not be found.
\\?\C:\Users\mark_\AppData\Local\atom\app-1.40.0\resources\app.asar.unpacked\node_modules\keyboard-layout\build\Release\keyboard-layout-manager.node

```
At ELECTRON_ASAR.js:166

Error: The specified module could not be found.
\\?\C:\Users\mark_\AppData\Local\atom\app-1.40.0\resources\app.asar.unpacked\node_modules\keyboard-layout\build\Release\keyboard-layout-manager.node
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:166:20)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:740:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:180:18)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:559:12)
    at Function.Module._load (internal/modules/cjs/loader.js:551:3)
    at Module.require (/app.asar/static/index.js:72:46)
    at require (internal/modules/cjs/helpers.js:20:18)
    at customRequire (~/AppData/Local/atom/app-1.40.0/resources/app/static/<embedded>:1:727319)
    at get_KeyboardLayoutManager (~/AppData/Local/atom/app-1.40.0/resources/app/static/<embedded>:14:2747076)
    at get_manager (~/AppData/Local/atom/app-1.40.0/resources/app/static/<embedded>:14:2747178)
    at Object.getCurrentKeyboardLayout (~/AppData/Local/atom/app-1.40.0/resources/app/static/<embedded>:14:2747304)
    at e.keystrokeForKeyboardEvent (~/AppData/Local/atom/app-1.40.0/resources/app/static/<embedded>:14:1087842)
    at KeymapManager.keystrokeForKeyboardEvent (~/AppData/Local/atom/app-1.40.0/resources/app/static/<embedded>:11:1246932)
    at KeymapManager.handleKeyboardEvent (~/AppData/Local/atom/app-1.40.0/resources/app/static/<embedded>:11:1244917)
    at WindowEventHandler.handleDocumentKeyEvent (~/AppData/Local/atom/app-1.40.0/resources/app/static/<embedded>:11:284329)
```

### Commands

```
```

### Non-Core Packages

```

```

**Comment by kuovonne** [comment]
I looked at some of the other issues and noticed that someone suggested running Atom in safe mode, so I tried to do that and I still get the big red error box whenever I press a key.

Here are the steps that I used:
1) Lookup the target of the Atom shortcut on my Windows desktop.
2) In the Windows taskbar, type "Command Prompt" in the search box.
3) In the Command Prompt window, change to the directory listed in the shortcut using the "CD" command.
4) In the Command Prompt window, type "atom --safe".
5) Wait for Atom to start.
6) Press a key on the keyboard.

**Comment by kuovonne** [comment]
I tried to take a screen capture of the big red error message, and I noticed that it now says that the issue has already been reported. My apologies if this is a duplicate reporting. However, I'm pretty sure that earlier the big red error message had a  "Create issue on atom/atom" button instead of the current "View Issue" button.

![image](https://user-images.githubusercontent.com/53977766/63035825-7258ed80-be81-11e9-8ef7-674c1f5ef7b1.png)

**Comment by kuovonne** [comment]
I looked in File Explorer trying to find the file that the error message describes, and of course I couldn't find it.

The error message references this file:
\?\C:\Users\mark_\AppData\Local\atom\app-1.40.0\resources\app.asar.unpacked\node_modules\keyboard-layout\build\Release\keyboard-layout-manager.node

I was able to find the "node_modules" folder, but not the "keyboard-layout" folder within it.
I also noticed that the "node_modules" folder for 1.40.0 has 27 items in it, but the corresponding folder for 1.39.1 has 33 items.

**Comment by rsese** [comment]
Thanks for contributing!

We noticed that this looks like a duplicate of #14461 so you can subscribe there for updates if you'd like.

To get past the error, I would recommend that you quit Atom and then try reinstalling it from https://atom.io.  If you have an anti-virus program running, temporarily disable it before reinstalling.

Let me know if that helps or if you still have trouble after that.

---

Because we treat our issues list as [the Atom team's backlog](https://en.wikipedia.org/wiki/Scrum_(software_development)#Product_backlog), we close duplicates to focus our work and not have to touch the same chunk of code for the same reason multiple times. This is also why we may mark something as duplicate that isn't an exact duplicate but is closely related.

For information on how to use GitHub's search feature to find out if something is a duplicate before filing, see [the How Can I Contribute? section of the Atom CONTRIBUTING guide](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-can-i-contribute).

**Comment by kuovonne** [comment]
Thank you so much for helping me out and being patient with a newbie! I apologize for not doing a search before.

I was able to reinstall Atom, and I no longer get the big red error message. Yay!

I looked in File Explorer, and I noticed that I no longer have a folder for 1.39.1 (or any of the other previous version numbers), only a folder for 1.40.0. The "node_modules" folder for 1.40.0 now contains 33 items (including the previously missing file) versus the 27 that it had before. 

I didn't record the specific items in the folder before, so I can't tell which folders changed, but the number now matches the number of items that were in the 1.39.1 version.
